src/biguint.rs

```diff
@@ -2285,6 +2285,9 @@ fn plain_modpow(base: &BigUint, exp_data: &[BigDigit], modulus: &BigUint) -> Big
 
     let mut exp_iter = exp_data[i + 1..].iter();
     if exp_iter.len() == 0 && r.is_one() {
+        if &base > modulus {
+            base %= modulus;
+        }
```

Comment:
I think it would be better to normalize this above where we create a `base.clone()`, just use `base % modulus` instead. The implementation of that `%` operator already checks if the value is less than the modulus too.